src/constraints/storage.jl

```diff
@@ -243,12 +249,45 @@ function add_storage_constraints!(connection, model, variables, expressions, con
 end
 
 function _append_storage_data_to_indices(connection, table_name)
+    join_duration = ""
+    select_duration = ""
+
+    if table_name == :balance_storage_over_clustered_year
+        DuckDB.query(
+            connection,
+            """
+            CREATE OR REPLACE TEMP TABLE t_duration_over_clustered_year AS
+            SELECT
+                con.asset,
+                con.year,
+                con.period_block_start,
+                SUM(mapping.num_timesteps) AS duration_period_block
+            FROM cons_balance_storage_over_clustered_year AS con
+            LEFT JOIN timeframe_data AS mapping
+                ON mapping.year = con.year
+                AND mapping.period BETWEEN con.period_block_start AND con.period_block_end
+            GROUP BY con.asset, con.year, con.period_block_start
+            ORDER BY con.asset, con.year, con.period_block_start
```

Comment:
```suggestion
                cons.asset,
                cons.year,
                cons.period_block_start,
                SUM(mapping.num_timesteps) AS duration_period_block
            FROM cons_balance_storage_over_clustered_year AS cons
            LEFT JOIN timeframe_data AS mapping
                ON mapping.year = cons.year
                AND mapping.period BETWEEN cons.period_block_start AND cons.period_block_end
            GROUP BY cons.asset, cons.year, cons.period_block_start
```

Changed `con` to `cons` for consistency with other parts (like below), and remove the ordering because this will be joined with another table, so it isn't needed.